test/decodeSchema.js

```diff
@@ -222,6 +222,21 @@ describe('ParquetSchema', function () {
     };
 
     const reader = new parquet.ParquetReader(metadata, {});
-    assert.deepEqual(reader.schema.fields, expected);
+    const removeUndefinedFields = (obj) => {
+      Object.keys(obj).forEach((key) => {
+        if (obj[key] === undefined || obj[key] === null || (Array.isArray(obj[key]) && obj[key].length === 0)) {
+          // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+          delete obj[key];
+        } else if (typeof obj[key] === 'object') {
+          removeUndefinedFields(obj[key]);
+        }
+      });
+      return obj;
+    };
+
+    const sanitizedExpected = removeUndefinedFields(expected);
+    const sanitizedActual = removeUndefinedFields(reader.schema.fields);
+
+    assert.deepEqual(sanitizedActual, sanitizedExpected);
```

Comment:
this will be reverted as i have to go in many files to add missing fields due to introduction of logical types

Reply:
this a test file which checks for correctness of schema with respect to a hardcoded expected result,  which are off now due to `logicalType` in field defintion hence the unit is failing due to mis-match 

I plan to remove it and update those fields